src/server/http/stats/handlers.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2024 The Tari Project
+// SPDX-License-Identifier: BSD-3-Clause
+
+use std::collections::HashMap;
+
+use axum::extract::State;
+use axum::http::StatusCode;
+use axum::Json;
+use itertools::Itertools;
+use log::error;
+use tari_common::configuration::Network;
+use tari_core::consensus::ConsensusManager;
+use tari_core::transactions::tari_amount::MicroMinotari;
+use tari_utilities::epoch_time::EpochTime;
+
+use crate::server::http::stats::models::{EstimatedEarnings, Stats};
+use crate::server::http::stats::server::AppState;
+use crate::sharechain::SHARE_COUNT;
+
+const LOG_TARGET: &str = "p2pool::server::stats::get";
+
+pub async fn handle_get_stats(State(state): State<AppState>) -> Result<Json<Stats>, StatusCode> {
+    let chain = state.share_chain.blocks(0).await.map_err(|error| {
+        error!(target: LOG_TARGET, "Failed to get blocks of share chain: {error:?}");
+        StatusCode::INTERNAL_SERVER_ERROR
+    })?;
+
+    // connected
+    let connected = state.peer_store.peer_count().await > 0;
+
+    // collect number of miners
+    let num_of_miners = chain
+        .iter()
+        .map(|block| block.miner_wallet_address())
+        .filter(|addr_opt| addr_opt.is_some())
+        .map(|addr| addr.as_ref().unwrap().to_base58())
+        .unique()
```

Comment:
q: when do we have duplicate addresses? if someone who previous left the pool re-joins?

Reply:
ah no, we are iterating over the current strongest chain, so as a miner could mine many blocks, then after mapping to wallet address we definitely will have duplicates, so thatswhy I needed `.unique()`